backend/Zeus.Api.Infrastructure/Authentication/JwtGenerator.cs

```diff
@@ -0,0 +1,59 @@
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+
+using Zeus.Api.Application.Common.Interfaces.Authentication;
+using Zeus.Api.Application.Common.Interfaces.Services;
+using Zeus.Api.Domain.Common.ValueObjects.Authentication;
+using Zeus.Api.Domain.UserAggregate;
+
+namespace Zeus.Api.Infrastructure.Authentication;
+
+public class JwtGenerator : IJwtGenerator
+{
+    private readonly IDateTimeProvider _dateTimeProvider;
+    private readonly JwtSettings _jwtSettings;
+
+    public JwtGenerator(IDateTimeProvider dateTimeProvider, IOptions<JwtSettings> jwtSettings)
+    {
+        _dateTimeProvider = dateTimeProvider;
+        _jwtSettings = jwtSettings.Value;
+    }
+
+    private string GenerateToken(User user)
+    {
+        var signingCredentials = new SigningCredentials(
+            new SymmetricSecurityKey(
+                Encoding.UTF8.GetBytes(_jwtSettings.Secret)),
+            SecurityAlgorithms.HmacSha256);
+        var claims = new[]
+        {
+            new Claim(JwtRegisteredClaimNames.Sub, user.Id.Value.ToString()),
+            new Claim(JwtRegisteredClaimNames.GivenName, user.FirstName),
+            new Claim(JwtRegisteredClaimNames.FamilyName, user.LastName),
+            new Claim(JwtRegisteredClaimNames.Jti, Guid.NewGuid().ToString()),
+        };
+
+        var securityToken = new JwtSecurityToken(
+            issuer: _jwtSettings.Issuer,
+            audience: _jwtSettings.Audience,
+            expires: _dateTimeProvider.UtcNow.AddMinutes(_jwtSettings.ExpiryMinutes),
+            claims: claims,
+            signingCredentials: signingCredentials);
+
+        return new JwtSecurityTokenHandler().WriteToken(securityToken);
+    }
+
+    public AccessToken GenerateAccessToken(User user)
+    {
+        return new AccessToken(GenerateToken(user));
```

Comment:
I think you should add a `Token.Type` in the claims to make difference between an `AccessToken` and a `RefreshToken` to avoid use one for the usage of an other. So, if you do it, you'll have to check at decoding time the type of each token.